src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAloneOrSingleLine2.java

```diff
@@ -0,0 +1,69 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyAloneOrSingleLine2 {
+
+    private int a;
+    private static int b;
+    {  a = 2; }
+    static { b = 3; }
+
+    void method1() {
+        Thread t = new Thread() {@Override public void run() {
+            int a; int b;} //violation
+        };
+    }
+
+    void method2(java.util.HashSet<String> set) {
+        java.util.Map<String, String> map1 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+            put("alpha", "betical");}  //violation
+        };
+
+        java.util.Map<String, String> map2 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+            put("alpha", "betical");
+        }};; //violation
```

Comment:
please remove extra ";"

`}};` is NOT a violation in case of "alone" -https://github.com/checkstyle/checkstyle/pull/4247/commits/1fed9f6b1f00a62cc3a4ef2aa0d317bb7d497c10#diff-18d61a16a1a99cd7aa45bcaa7b217826R29
it is not violation for same mode and code at https://github.com/checkstyle/checkstyle/pull/4247/commits/1fed9f6b1f00a62cc3a4ef2aa0d317bb7d497c10#diff-a7e700698e09368ce7408af88bf91357R108

 ... Why it is a violation in case "alone_or_singleline" ?
behavior should be the same with extra allowances for singleline cases.

Reply:
So, we don't want a violation for `alone` and `alone_or_singleline` for such code correct ?